### ISSUE TITLE
Willing to incorporate changes adding support for additional Handlebars features to parser v1?

### ISSUE THREAD

**Issue by j-f1** [glimmerjs/glimmer-vm#1367]
Hello! I’m working on improving the Handlebars support in [Prettier](https://prettier.io), which uses `@glimmer/syntax` (specifically, the `preprocess` function) to parse `.hbs` files. As you’re probably aware, Glimmer doesn’t support all Handlebars features. However, it wasn’t that difficult to add support for parsing them correctly, at least in the first pass. Would you be willing to accept a PR that adds support for the missing features (including decorators, partials, and slash paths) to the first level of the parser (perhaps via an option, or by throwing an error in the `v2-a` level of the parser, to match the current behavior)?

**Comment by j-f1** [comment]
My fork with the changes (which include a removal of the packages Prettier doesn’t need) is here: https://github.com/j-f1/forked-glimmer-vm/tree/prettier

**Comment by rwjblue** [comment]
I think in general, adding support for things in `@glimmer/syntax` is fine as long as we ensure that `@glimmer/compiler` properly errors for those constructs.

**Comment by websolutions-hamburg** [comment]
Is there any news on this?

**Comment by j-f1** [comment]
No, I ultimately ended up realizing that getting formatting for handlebars to actually work correctly for most/all input that produces valid output is basically impossible. As a result I switched my project to use a different template language and so gave up on the changes to Glimmer. Sorry I don’t have anything more positive :(
